feature/interface/singleton/ate_tests/singleton_test/singleton_test.go

```diff
@@ -400,19 +434,62 @@ func (tc *testCase) run(t *testing.T) {
 	}
 }
 
-func TestSingleton(t *testing.T) {
+func TestMTUs(t *testing.T) {
 	dut := ondatra.DUT(t, "dut")
 	ate := ondatra.ATE(t, "ate")
 
+	// These are the L3 MTUs, i.e. the payload portion of an Ethernet frame.
 	mtus := []uint16{1500, 5000, 9212}
+
 	for _, mtu := range mtus {
 		top := ate.Topology().New()
 		tc := &testCase{
-			mtu: mtu,
+			mtu:  mtu,
+			auto: forcedNegotiation,
+
 			dut: dut,
 			ate: ate,
 			top: top,
 		}
-		t.Run(tc.string(), tc.run)
+		t.Run(fmt.Sprintf("MTU=%d", mtu), tc.testMTU)
+	}
+}
+
+var autoModeName = map[autoMode]string{
+	forcedNegotiation:              "Forced",
+	autoNegotiation:                "Auto",
+	autoNegotiationWithDuplexSpeed: "AutoWithDuplexSpeed",
+}
+
+// TestNegotiate validates that port speed is reported correctly and that port telemetry
+// atches expected negotiated speeds for forced, auto-negotiation, and auto-negotiation
+// while overriding port speed and duplex.
+func TestNegotiate(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	ate := ondatra.ATE(t, "ate")
+
+	for auto, name := range autoModeName {
+		t.Run(name, func(t *testing.T) {
+			top := ate.Topology().New()
+			tc := &testCase{
+				mtu:  1500,
```

Comment:
nit: some comment on this magic 1500 number? (e.g. // fix the L3 MTU to 1500 because this test focus on auto negotiation).

Reply:
Seems like you're able to figure this out without the explanation, so it goes without saying. 😀